apps/remote_control/test/lexical/remote_control/code_mod/ast/aliases_test.exs

```diff
@@ -0,0 +1,233 @@
+    doc = Document.new("file:///file.ex", text, 0)
+    at(doc, pos)
+  end
+
+  describe "top level aliases" do
+    test "an alias outside of a module" do
+      aliases =
+        ~q[
+        alias Foo.Bar.Baz
+        defmodule Parent do
+          |
+        end
+      ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Baz] == Foo.Bar.Baz
+    end
+
+    test "an alias inside the body of a module" do
+      aliases =
+        ~q[
+          defmodule Basic do
+            alias Foo.Bar
+            |
+          end
+          ]
+        |> aliases_at_cursor()
+
+      assert aliases == %{__MODULE__: Basic, Bar: Foo.Bar}
+    end
+
+    test "an alias using as" do
+      aliases =
+        ~q[
+          defmodule TopLevel do
+            alias Foo.Bar, as: FooBar
+            |
+          end
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:__MODULE__] == TopLevel
+      assert aliases[:FooBar] == Foo.Bar
+    end
+
+    test "multiple aliases" do
+      aliases =
+        ~q[
+          defmodule TopLevel do
+            alias Foo.Bar.{First, Second, Third.Fourth}
+            |
+          end
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:First] == Foo.Bar.First
+      assert aliases[:Second] == Foo.Bar.Second
+      assert aliases[:Fourth] == Foo.Bar.Third.Fourth
+    end
+
+    test "aliases expanding other aliases" do
+      aliases =
+        ~q[
+          alias Foo.Bar.Baz
+          alias Baz.Quux|
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Baz] == Foo.Bar.Baz
+      assert aliases[:Quux] == Foo.Bar.Baz.Quux
+    end
+
+    test "allows overrides" do
+      aliases =
+        ~q[
+          alias Foo.Bar.Baz
+          alias Other.Baz
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Baz] == Other.Baz
+    end
+  end
+
+  describe "alias scopes" do
+    test "aliases are removed when leaving a module" do
+      aliases =
+        ~q[
+          defmodule Basic do
+            alias Foo.Bar
+          end|
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases == %{Basic: Basic}
+    end
+
+    test "aliases inside of nested modules" do
+      aliases =
+        ~q[
+          defmodule Parent do
+            alias Foo.Grandparent
+
+            defmodule Child do
+              alias Foo.Something
+              |
+            end
+          end
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Grandparent] == Foo.Grandparent
+      assert aliases[:__MODULE__] == Parent.Child
+      assert aliases[:Something] == Foo.Something
+    end
+
+    test "a nested module is not aliased inside the module" do
+      aliases =
+        ~q[
+          defmodule Parent do
+            alias Foo.Grandparent
+
+            defmodule Child do
+              alias Foo.Something
+              |
+            end
+
+          end
+        ]
+        |> aliases_at_cursor()
+
+      refute aliases[:Child]
```

Comment:
but how to complete this case:

```elixir
defmodule Parent do
  defmodule Child do
    defstruct [:a]

    def foo(%Child{|}) do
    end
  end
end
```

I think `aliases[:Child]` should be `Parent.Child`

Reply:
no, you cannot do this; the `Child` module hasn't been defined yet, you're still inside it. You need to use `__MODULE__`.

Reply:
Hmm, this behavior has changed, though I'd still caution against using the name of the module _in_ the module, it's not refactor friendly.